harness/determined/searcher/_search_method.py

```diff
@@ -261,6 +410,39 @@ def on_trial_exited_early(
     ) -> List[Operation]:
         """
         Informs the searcher that a trial has exited earlier than expected.
+
+        Example:
+
+        .. code:: python
+
+            def on_trial_exited_early(
+                self,
+                searcher_state: SearcherState,
+                request_id: uuid.UUID,
+                exited_reason: ExitedReason,
+            ) -> List[Operation]:
+                MAX_FAILURES = 10  # Best defined in __init__
```

Comment:
Personal preference, but the various globals like `MAX_FAILURES` seem a little unexpected. Would instead expect simple attributes: `self.max_failures` and similar.

Reply:
Added